modules/switches.py

```diff
@@ -0,0 +1,93 @@
+import random
+import modules
+
+
+class Switches(modules.Module):
+    identifiers = ['switches']
+    display_name = "Switches"
+    manual_name = "Switches"
+    help_text = "`{cmd} flip 1 5 3 4 2 5 2` or `{cmd} flip 1534252`. Partial solutions allowed."
+    module_score = 1
+
+    def __init__(self, bomb, ident):
+        super().__init__(bomb, ident)
+
+        self.invalidPositions = "11011/10100/10000/01101/01100/01000/00111/00101/00011/00001".split("/")
+
+        self.solution = "10000"
+        while self.solution in self.invalidPositions:
+            self.solution = ''.join([str(random.randint(0, 1)) for i in range(5)])
+
+        self.switches = "10000"
+        while self.switches in self.invalidPositions or self.switches == self.solution:
+            self.switches = ''.join([str(random.randint(0, 1)) for i in range(5)])
+
+        self.positionNames = ["up", "down"]
+
+        self.log(f"Solution: {' '.join(map(lambda x:self.positionNames[int(x)],list(self.solution)))}")
+
+    def generateSwitch(self, index):
+        default = [[50.5, 174], [65.5, 174], [73, 234], [43, 234]]
+        position = list(map(lambda x: [x[0] + 58 * index, x[1]], default))
+        if self.switches[index] == "0":
+            position[2][1] = 348 - position[2][1]
+            position[3][1] = 348 - position[3][1]
+
+        position = map(lambda x: map(lambda y: str(y), x), position)
+
+        o = f'<circle stroke-width="2" stroke="#000" cx="{58*(index+1)}" cy="174" r="15" />'
+        o += f'<circle stroke-width="2" stroke="#000" fill="{"#0f0" if self.solution[index]=="0" else "#fff"}" cx="{58*(index+1)}" cy="87" r="10" />'
+        o += f'<circle stroke-width="2" stroke="#000" fill="{"#0f0" if self.solution[index]=="1" else "#fff"}" cx="{58*(index+1)}" cy="261" r="10" />'
+        o += f'<polygon points="{" ".join(list(map(lambda x:",".join(x),position)))}" style="fill:black;stroke:black;stroke-width:2" />'
```

Comment:
It's not necessary to use `list()` here.

Reply:
oh whoops I thought I removed all of those